temporal-sdk/src/main/java/io/temporal/worker/tuning/SlotSupplier.java

```diff
@@ -36,16 +37,19 @@
 @Experimental
 public interface SlotSupplier<SI extends SlotInfo> {
   /**
-   * This function is called before polling for new tasks. Your implementation should block until a
-   * slot is available then return a permit to use that slot.
+   * This function is called before polling for new tasks. Your implementation should return a
+   * Promise that is completed with a {@link SlotPermit} when one becomes available.
+   *
+   * <p>These futures may be cancelled if the worker is shutting down or otherwise abandons the
```

Comment:
>These futures may be cancelled if the worker is shutting down or otherwise abandons the

You are aware `CompletableFuture` cancellation does not propagate upstream correct?